// @datadog/datadog-api-client/dist/packages/datadog-api-client-v1/models/AzureAccountMetricsConfig.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.AzureAccountMetricsConfig = void 0;
/**
 * Dictionary containing the key `excluded_resource_providers` which has to be a list of Microsoft Azure Resource Provider names.
 * This feature is currently being beta tested.
 * In order to enable all resource providers for metric collection, pass:
 * `metrics_config: {"excluded_resource_providers": []}` (i.e., an empty list for `excluded_resource_providers`).
 */
class AzureAccountMetricsConfig {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return AzureAccountMetricsConfig.attributeTypeMap;
    }
}
exports.AzureAccountMetricsConfig = AzureAccountMetricsConfig;
/**
 * @ignore
 */
AzureAccountMetricsConfig.attributeTypeMap = {
    excludedResourceProviders: {
        baseName: "excluded_resource_providers",
        type: "Array<string>",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "any",
    },
};
