// @datadog/datadog-api-client/dist/packages/datadog-api-client-v1/models/SharedDashboardUpdateRequest.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.SharedDashboardUpdateRequest = void 0;
/**
 * Update a shared dashboard's settings.
 */
class SharedDashboardUpdateRequest {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return SharedDashboardUpdateRequest.attributeTypeMap;
    }
}
exports.SharedDashboardUpdateRequest = SharedDashboardUpdateRequest;
/**
 * @ignore
 */
SharedDashboardUpdateRequest.attributeTypeMap = {
    globalTime: {
        baseName: "global_time",
        type: "SharedDashboardUpdateRequestGlobalTime",
        required: true,
    },
    globalTimeSelectableEnabled: {
        baseName: "global_time_selectable_enabled",
        type: "boolean",
    },
    selectableTemplateVars: {
        baseName: "selectable_template_vars",
        type: "Array<SelectableTemplateVariableItems>",
    },
    shareList: {
        baseName: "share_list",
        type: "Array<string>",
    },
    shareType: {
        baseName: "share_type",
        type: "DashboardShareType",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "any",
    },
};
