// @datadog/datadog-api-client/dist/packages/datadog-api-client-v1/models/SyntheticsMobileTestOptions.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.SyntheticsMobileTestOptions = void 0;
/**
 * Object describing the extra options for a Synthetic test.
 */
class SyntheticsMobileTestOptions {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return SyntheticsMobileTestOptions.attributeTypeMap;
    }
}
exports.SyntheticsMobileTestOptions = SyntheticsMobileTestOptions;
/**
 * @ignore
 */
SyntheticsMobileTestOptions.attributeTypeMap = {
    allowApplicationCrash: {
        baseName: "allowApplicationCrash",
        type: "boolean",
    },
    bindings: {
        baseName: "bindings",
        type: "Array<SyntheticsMobileTestBinding>",
    },
    ci: {
        baseName: "ci",
        type: "SyntheticsMobileTestCiOptions",
    },
    defaultStepTimeout: {
        baseName: "defaultStepTimeout",
        type: "number",
        format: "int32",
    },
    deviceIds: {
        baseName: "device_ids",
        type: "Array<SyntheticsMobileDeviceID>",
    },
    disableAutoAcceptAlert: {
        baseName: "disableAutoAcceptAlert",
        type: "boolean",
    },
    minFailureDuration: {
        baseName: "min_failure_duration",
        type: "number",
        format: "int64",
    },
    mobileApplication: {
        baseName: "mobileApplication",
        type: "SyntheticsMobileTestsMobileApplication",
    },
    monitorName: {
        baseName: "monitor_name",
        type: "string",
    },
    monitorOptions: {
        baseName: "monitor_options",
        type: "SyntheticsMobileTestOptionsMonitorOptions",
    },
    monitorPriority: {
        baseName: "monitor_priority",
        type: "number",
        format: "int32",
    },
    noScreenshot: {
        baseName: "noScreenshot",
        type: "boolean",
    },
    restrictedRoles: {
        baseName: "restricted_roles",
        type: "Array<string>",
    },
    retry: {
        baseName: "retry",
        type: "SyntheticsTestOptionsRetry",
    },
    scheduling: {
        baseName: "scheduling",
        type: "SyntheticsTestOptionsScheduling",
    },
    tickEvery: {
        baseName: "tick_every",
        type: "number",
        format: "int64",
    },
    verbosity: {
        baseName: "verbosity",
        type: "number",
        format: "int32",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "any",
    },
};
