// @datadog/datadog-api-client/dist/packages/datadog-api-client-v1/models/SyntheticsMobileTestOptionsMonitorOptions.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.SyntheticsMobileTestOptionsMonitorOptions = void 0;
/**
 * Object containing the options for a mobile Synthetic test as a monitor
 * (for example, renotification).
 */
class SyntheticsMobileTestOptionsMonitorOptions {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return SyntheticsMobileTestOptionsMonitorOptions.attributeTypeMap;
    }
}
exports.SyntheticsMobileTestOptionsMonitorOptions = SyntheticsMobileTestOptionsMonitorOptions;
/**
 * @ignore
 */
SyntheticsMobileTestOptionsMonitorOptions.attributeTypeMap = {
    escalationMessage: {
        baseName: "escalation_message",
        type: "string",
    },
    notificationPresetName: {
        baseName: "notification_preset_name",
        type: "SyntheticsMobileTestOptionsMonitorOptionsNotificationPresetName",
    },
    renotifyInterval: {
        baseName: "renotify_interval",
        type: "number",
        format: "int64",
    },
    renotifyOccurrences: {
        baseName: "renotify_occurrences",
        type: "number",
        format: "int64",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "any",
    },
};
