// @datadog/datadog-api-client/dist/packages/datadog-api-client-v1/models/SyntheticsMobileTestInitialApplicationArgumentsPropertyNames.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.SyntheticsMobileTestInitialApplicationArgumentsPropertyNames = void 0;
/**
 * Name of the property.
 */
class SyntheticsMobileTestInitialApplicationArgumentsPropertyNames {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return SyntheticsMobileTestInitialApplicationArgumentsPropertyNames.attributeTypeMap;
    }
}
exports.SyntheticsMobileTestInitialApplicationArgumentsPropertyNames = SyntheticsMobileTestInitialApplicationArgumentsPropertyNames;
/**
 * @ignore
 */
SyntheticsMobileTestInitialApplicationArgumentsPropertyNames.attributeTypeMap = {
    pattern: {
        baseName: "pattern",
        type: "string",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "any",
    },
};
