// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/AWSRelatedAccountAttributes.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.AWSRelatedAccountAttributes = void 0;
/**
 * Attributes for an AWS related account.
 */
class AWSRelatedAccountAttributes {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return AWSRelatedAccountAttributes.attributeTypeMap;
    }
}
exports.AWSRelatedAccountAttributes = AWSRelatedAccountAttributes;
/**
 * @ignore
 */
AWSRelatedAccountAttributes.attributeTypeMap = {
    hasDatadogIntegration: {
        baseName: "has_datadog_integration",
        type: "boolean",
    },
    name: {
        baseName: "name",
        type: "string",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "any",
    },
};
